<robot name="test_robot">
    <link name="base_link">
        <visual>
            <geometry>
                <box size="0.16 0.16 0.12"/>
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0.085"/>
            <material name="white">
                <color rgba="1 1 1 1"/>
            </material>
        </visual>
        <collision>
            <geometry>
                <box size="0.16 0.16 0.12"/>
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0.085"/>
        </collision>
    </link>

    <link name="left_wheel">
        <visual>
            <geometry>
                <cylinder length="0.008" radius="0.046"/>
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0"/>
            <material name="black">
                <color rgba="0 0 0 1"/>
            </material>
        </visual>
        <collision>
            <geometry>
                <cylinder length="0.008" radius="0.046"/>
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0"/>
        </collision>
    </link>

    <joint name="base_to_left_wheel" type="continuous">
        <parent link="base_link"/>
        <child link="left_wheel"/>
        <origin xyz="0 0.115 0.046" rpy="1.5708 0 0"/>
        <axis xyz="0 0 1"/>
    </joint>

    <link name="right_wheel">
        <visual>
            <geometry>
                <cylinder length="0.008" radius="0.046"/>
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0"/>
            <material name="black">
                <color rgba="0 0 0 1"/>
            </material>
        </visual>
        <collision>
            <geometry>
                <cylinder length="0.008" radius="0.046"/>
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0"/>
        </collision>
    </link>

    <joint name="base_to_right_wheel" type="continuous">
        <parent link="base_link"/>
        <child link="right_wheel"/>
        <origin xyz="0 -0.115 0.046" rpy="-1.5708 0 0"/>
        <axis xyz="0 0 1"/>
    </joint>

    <link name="front_wheel_attach">
        <visual>
            <geometry>
                <cylinder length="0.060" radius="0.023"/>
            </geometry>
            <origin rpy="0 0 0" xyz="0.13 0 0.08"/>
            <material name="white">
                <color rgba="1 1 1 1"/>
            </material>
        </visual>
    </link>

    <joint name="base_to_front_attach" type="fixed">
        <parent link="base_link"/>
        <child link="front_wheel_attach"/>
        <origin xyz="0 0 0" rpy="0 0 0"/>
    </joint>

    <link name="front_wheel">
        <visual>
            <geometry>
                <sphere radius="0.020"/>
            </geometry>
            <origin rpy="1.578 0 0" xyz="0 0 0"/>
            <material name="black">
                <color rgba="0 0 0 1"/>
            </material>
        </visual>
        <collision>
            <geometry>
                <sphere radius="0.020"/>
            </geometry>
            <origin rpy="1.578 0 0" xyz="0 0 0"/>
        </collision>
    </link>

    <joint name="base_to_front_wheel" type="fixed">
        <parent link="base_link"/>
        <child link="front_wheel"/>
        <origin xyz="0.13 0 0.02" rpy="0 0 0"/>
    </joint>
</robot>
